modules/costs/app/contracts/time_entries/create_contract.rb

```diff
@@ -29,18 +29,24 @@
 module TimeEntries
   class CreateContract < BaseContract
     validate :user_allowed_to_add
-    validate :validate_user_current_user
 
     private
 
     def user_allowed_to_add
-      if model.project && !user.allowed_to?(:log_time, model.project)
-        errors.add :base, :error_unauthorized
+      if model.project && model.user
+        return if allowed_to_log_for_others?
+        return if allowed_to_log_to_himself?
       end
+
+      errors.add :base, :error_unauthorized
```

Comment:
I think this is misformulated. Currently, whenever the time entry lacks either the user or the project an `error_unauthorized` is thrown, which explains the failing request and I'd assume also the feature specs.

Changing it to:
```
    def user_allowed_to_add
       return unless model.project && model.user
       return if allowed_to_log_for_others?
       return if allowed_to_log_to_himself?
       
       errors.add :base, :error_unauthorized
    end   
```   

should work